### PR TITLE
refactor(rust): unify the way of creating symbols

### PR DESCRIPTION
**Pull request by hyf0**

### CODE REVIEW

**Comment by hyf0** [comment]
* **#2449** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2449?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2448** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2448?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2447** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2447?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2446** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/2446?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @hyf0 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/rolldown_common/src/types/symbol_ref_db.rs [hunk 1]
```diff
@@ -60,21 +64,30 @@ impl SymbolRefDbForModule {
...
 
-  pub fn create_symbol(
-    &mut self,
-    span: Span,
-    name: CompactString,
-    flags: SymbolFlags,
-    scope_id: ScopeId,
-    node_id: NodeId,
-  ) -> SymbolId {
+  // The `facade` means the symbol is actually not exist in the AST.
```
Comment (IWANABETHATGUY) [inline crates/rolldown_common/src/types/symbol_ref_db.rs:67]: `facade` not have `not exists` meaning in English, I prefer `dummy`

Comment (IWANABETHATGUY) [reply crates/rolldown_common/src/types/symbol_ref_db.rs:67]: you could refer here https://github.com/rolldown/rolldown/pull/2447/files#diff-a572f77d3befed0f4403b2a59164bd5179e5daa68ac58d2c74597e5fbc68e563R80

Comment (IWANABETHATGUY) [reply crates/rolldown_common/src/types/symbol_ref_db.rs:67]: > `facade` not have `not exists` meaning in English, I prefer `dummy`

Or say, it did not widely used for meaning `fake node` in cs, that's why I recommend `dummy`, just like `oxc` did

**Comment by hyf0** [comment]
### Merge activity

* **Oct 17, 2:50 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/rolldown/rolldown/2447) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

**Review by IWANABETHATGUY — APPROVED** [review]

**Pull request merged**